modules/nw-ovn-ipsec-verification.adoc

```diff
@@ -0,0 +1,280 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/about-ipsec-ovn.adoc
+
+[id="nw-ovn-ipsec-verification_{context}"]
+= Verifying that IPsec is enabled
+
+As a cluster administrator, you can verify that IPsec is enabled.
+
+.Verification
+
+. To find the names of the IPsec pods, enter the following command:
++
+[source,terminal]
+----
+$ oc get pods -n openshift-ovn-kubernetes | grep ipsec
+----
++
+.Example output
+[source,terminal]
+----
+ovn-ipsec-4496s        1/1     Running   0          6h39m
+ovn-ipsec-d6cht        1/1     Running   0          6h42m
+ovn-ipsec-skblc        1/1     Running   0          6h51m
+ovn-ipsec-vf8rf        1/1     Running   0          6h51m
+ovn-ipsec-w7hjr        1/1     Running   0          6h51m
+ovn-ipsec-zsk7x        1/1     Running   0          6h42m
+----
+
+. Verify that IPsec tunnels are established on your cluster:
++
+[source,terminal]
+----
+$ oc -n openshift-ovn-kubernetes -c ovn-ipsec \
+  exec <ipsec_pod> -- ovs-appctl -t ovs-monitor-ipsec tunnels/show
+----
++
+--
+where:
+
+`<ipsec_pod>`:: Specifies an IPsec pod from the first step.
```

Comment:
I wonder if this low-level information is too much. Maybe they could just look through the pod logs for errors?

Reply:
@markdgray, good idea. Is this output enough for validation? Thanks!

```
$ oc logs pod/ovn-ipsec-46zfz -n openshift-ovn-kubernetes

+ trap cleanup SIGTERM
+ counter=0
ovnkube-node has configured node.
+ '[' -f /etc/cni/net.d/10-ovn-kubernetes.conf ']'
+ echo 'ovnkube-node has configured node.'
+ ip x s flush
+ ip x p flush
+ ulimit -n 1024
+ /usr/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig
+ /usr/libexec/ipsec/_stackmanager start
+ /usr/sbin/ipsec --checknss
Initializing NSS database

+ /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --logfile /var/log/openvswitch/libreswan.log
+ OVS_LOGDIR=/var/log/openvswitch
+ OVS_RUNDIR=/var/run/openvswitch
+ OVS_PKGDATADIR=/usr/share/openvswitch
+ /usr/share/openvswitch/scripts/ovs-ctl --ike-daemon=libreswan --no-restart-ike-daemon start-ovs-ipsec
+ sleep infinity
```

Reply:
I just had a think about this. It is probably better that we don't share too much internal information as it is subject to change. Probably the simplest method to confirm that it is enabled is:

`oc -n openshift-ovn-kubernetes rsh -c nbdb ovnkube-master-ggppg  ovn-nbctl --no-leader-only get nb_global . ipsec`

This will return `true` if enabled.

Even presence of the pods (in the current version) does not guarantee it is enabled.

Reply:
@markdgray, awesome! I've updated the procedure.